### PR TITLE
1474443 compliance.created events now use UUID for the 'consumerId' f…

### PR DESCRIPTION
**Pull request by nikosmoum**
…ield instead of an internal db id

### CODE REVIEW

**Comment by nikosmoum** [comment]
[1474443 - compliance.created messages use internal id for 'consumerId'](https://trello.com/c/QnOOT1lg/588-1474443-compliancecreated-messages-use-internal-id-for-consumerid)

File: server/src/main/java/org/candlepin/audit/EventFactory.java [hunk 1]
```diff
@@ -233,7 +233,7 @@ public Event subscriptionExpired(Subscription subscription) {
     public Event complianceCreated(Consumer consumer,
         Set<Entitlement> entitlements, ComplianceStatus compliance) {
         return new Event(Event.Type.CREATED, Event.Target.COMPLIANCE,
-            consumer.getName(), principalProvider.get(), consumer.getOwner().getId(), consumer.getId(),
+            consumer.getName(), principalProvider.get(), consumer.getOwner().getId(), consumer.getUuid(),
             consumer.getId(), null, buildComplianceDataJson(consumer, entitlements, compliance), null,
```
Comment (cnsnyder) [inline server/src/main/java/org/candlepin/audit/EventFactory.java]: We'll need the second usage of consumer.getId() replaced with consumer.getUuid() as well.

**Comment by Ceiu** [comment]
The change itself is fine, but a little odd. Definitely worth adding a comment explaining why we're using the UUID in place of the ID; especially since we can't just change the JSON output of the event at the moment.

Also, has any investigation been done about removing the Consumer object itself from the output? Such a change may coincide with a larger event rework, but we should start doing the investigation now as we're making changes here.

**Comment by nikosmoum** [comment]
There has been some investigation by Barnaby I believe, with details in this [document](https://docs.google.com/document/d/1qcucpyMWpkg-b3Zi0hNDzPBCJYbn0RUbqcp4_2dW4eg/edit#), about Katello using only few specific fields, with part of the larger plan being to remove the Consumer object.

**Comment by nikosmoum** [comment]
@cnsnyder I have altered the second call to use getUuid() too, and amended my original commit message.

**Comment by barnabycourt** [comment]
@nikosmoum Please squash the two commits to one.

**Comment by Ceiu** [comment]
ACK

**Pull request merged**